NEW: Send Notification Without File when E-mail Template doesn't attach file

almisbahinfo opened the pull request:
When we create an Email Template that doesn't require main file to be attached, we still send notification with an attached file, eg: when we send a notification when validating proposal it send proposal with attached file with mention NOT VALIDATED.
I added OR condition on checking pdf_path.

almisbahinfo committed:
Send Notification Without File when E-mail Template does'nt attach file

When we create an Email Template that does'nt require main file to be attached, we still send notification with an attached file, eg: when we send a notification when validating proposal it send proposal with attached file with mention NOT VALIDATED.
I added OR condition on checking pdf_path.

eldy commented on htdocs/core/class/notify.class.php:
```diff
@@ -615,7 +615,7 @@ public function send($notifcode, $object, $filename_list = array(), $mimetype_li
 						$ref = dol_sanitizeFileName($newref);
 						$pdf_path = $dir_output."/".$ref.".pdf";
-						if (!dol_is_file($pdf_path)) {
+						if (!dol_is_file($pdf_path)||!$arraydefaultmessage->joinfiles) {
```
$arraydefaultmessage may still be null here (it depends on context). And if null, $arraydefaultmessage->joinfiles will generate a fatal error. So test must be enhanced to support this case...

almisbahinfo committed:
FIX #21141 Update condition to check for attached file to E-mail Template

Update condition to check for attached file to E-ail Template, initial Commit doesn't check for  if null (may generate fatal error depending on context), now we check if is an object, before checking if we join files.
